tests/parser/EclipseGridTests.cpp

```diff
@@ -2148,6 +2148,64 @@ BOOST_AUTO_TEST_CASE(TESTCP_ACTNUM_UPDATE) {
 }
 
 
+BOOST_AUTO_TEST_CASE(TESTCP_ACTNUM_AQUNUM) {
+    const char* deckData =
+
+        "RUNSPEC\n"
+        "\n"
+        "DIMENS\n"
+        " 3 2 1 /\n"
+        "GRID\n"
+        "COORD\n"
+        " 2000.0000  2000.0000  2000.0000   1999.9476  2000.0000  2002.9995\n"
+        " 2049.9924  2000.0000  2000.8726   2049.9400  2000.0000  2003.8722 \n"
+        " 2099.9848  2000.0000  2001.7452   2099.9324  2000.0000  2004.7448 \n"
+        " 2149.9772  2000.0000  2002.6179   2149.9248  2000.0000  2005.6174 \n"
+        " 2000.0000  2050.0000  2000.0000   1999.9476  2050.0000  2002.9995 \n"
+        " 2049.9924  2050.0000  2000.8726   2049.9400  2050.0000  2003.8722 \n"
+        " 2099.9848  2050.0000  2001.7452   2099.9324  2050.0000  2004.7448 \n"
+        " 2149.9772  2050.0000  2002.6179   2149.9248  2050.0000  2005.6174 \n"
+        " 2000.0000  2100.0000  2000.0000   1999.9476  2100.0000  2002.9995 \n"
+        " 2049.9924  2100.0000  2000.8726   2049.9400  2100.0000  2003.8722 \n"
+        " 2099.9848  2100.0000  2001.7452   2099.9324  2100.0000  2004.7448 \n"
+        " 2149.9772  2100.0000  2002.6179   2149.9248  2100.0000  2005.6174 / \n"
+        "ZCORN\n"
+        " 2000.0000  2000.8726  2000.8726  2001.7452  2001.7452  2002.6179 \n"
+        " 2000.0000  2000.8726  2000.8726  2001.7452  2001.7452  2002.6179 \n"
+        " 2000.0000  2000.8726  2000.8726  2001.7452  2001.7452  2002.6179 \n"
+        " 2000.0000  2000.8726  2000.8726  2001.7452  2001.7452  2002.6179 \n"
+        " 2002.9995  2003.8722  2003.8722  2004.7448  2004.7448  2005.6174 \n"
+        " 2002.9995  2003.8722  2003.8722  2004.7448  2004.7448  2005.6174 \n"
+        " 2002.9995  2003.8722  2003.8722  2004.7448  2004.7448  2005.6174 \n"
+        " 2002.9995  2003.8722  2003.8722  2004.7448  2004.7448  2005.6174 / \n"
+        "ACTNUM\n"
+        " 0 0 1 1 0 1 / \n"
+        "PORO\n"
+        "  6*0.15 /\n"
+        "AQUNUM\n"
+        "    1     1  1  1   1000000.0   10000   0.25   400    2585.00   285.00    1   1  /\n"
+        "    1     2  1  1   1000000.0   10000   0.25   400    2585.00   285.00    1   1  /\n"
+        "/\n"
+        "AQUCON\n"
+        "   1     3    3      2    2     1   1   'J+'      1.00      1  /\n"
+        "/\n"
+        "EDIT\n"
+        "\n";
+
+    Opm::Parser parser;
+    const auto deck = parser.parseString( deckData) ;
+    Opm::EclipseGrid grid( deck);
```

Comment:
In addition to this testing you should include a test which creates an `EclipseState` and verify that also that grid is correct:
```C++
Opm::EclipseState es(deck):
const auto& grid2 = es.getInputGrid()
...
BOOST_CHECK( desired_actnum == grid2.getACTNUM() )
```

Reply:
You are right, it reset back again. I am checking how that happens.

Reply:
It happens inn the `EclipseState` constructor

Reply:
Big question here is,  `FieldProps` has the member `m_actnum`, why we still have `ACTNUM` in `int_data`. 

`fp->actnum()` function will still check the `int_data["ACTNUM"]`.  What is the logic here?  Basically, it is `int_data["ACTNUM"]` discards the change we made through `AQUNUM`. 

```
std::vector<int> FieldProps::actnum() {
    auto actnum = this->m_actnum;
    const auto& deck_actnum = this->init_get<int>("ACTNUM");

    std::vector<int> global_map(this->active_size);
    {
        std::size_t active_index = 0;
        for (std::size_t g = 0; g < this->global_size; g++) {
            if (this->m_actnum[g]) {
                global_map[active_index] = g;
                active_index++;
            }
        }
    }
```

Reply:
```
    const auto& deck_actnum = this->init_get<int>("ACTNUM");
```
Does not have the full `ACTNUM`. It only has the value from active cells from `m_actnum`.

Reply:
Maybe it is for `BOX` or `EQUAL` operations?